src/cursor/cur_version.c

```diff
@@ -0,0 +1,453 @@
+    table_cursor = version_cursor->table_cursor;
+    CURSOR_API_CALL(cursor, session, close, NULL);
+err:
+    version_cursor->next_upd = NULL;
+    if (table_cursor != NULL)
+        WT_TRET(table_cursor->close(table_cursor));
+    if (hs_cursor != NULL)
+        WT_TRET(hs_cursor->close(hs_cursor));
+    __wt_cursor_close(cursor);
+
+    API_END_RET(session, ret);
+}
+
+/*
+ * __wt_curversion_open --
+ *     Initialize a version cursor.
+ */
+int
+__wt_curversion_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner, const char *cfg[],
+  WT_CURSOR **cursorp)
+{
+    WT_CURSOR_STATIC_INIT(iface, __curversion_get_key, /* get-key */
+      __wt_cursor_get_value,                           /* get-value */
+      __curversion_set_key,                            /* set-key */
+      __wt_cursor_set_value_notsup,                    /* set-value */
+      __wt_cursor_compare_notsup,                      /* compare */
+      __wt_cursor_equals_notsup,                       /* equals */
+      __curversion_next,                               /* next */
+      __wt_cursor_notsup,                              /* prev */
+      __curversion_reset,                              /* reset */
+      __curversion_search,                             /* search */
+      __wt_cursor_search_near_notsup,                  /* search-near */
+      __wt_cursor_notsup,                              /* insert */
+      __wt_cursor_modify_notsup,                       /* modify */
+      __wt_cursor_notsup,                              /* update */
+      __wt_cursor_notsup,                              /* remove */
+      __wt_cursor_notsup,                              /* reserve */
+      __wt_cursor_reconfigure_notsup,                  /* reconfigure */
+      __wt_cursor_notsup,                              /* largest_key */
+      __wt_cursor_notsup,                              /* cache */
+      __wt_cursor_reopen_notsup,                       /* reopen */
+      __curversion_close);                             /* close */
+
+    WT_CURSOR *cursor;
+    WT_CURSOR_VERSION *version_cursor;
+    WT_DECL_RET;
+    /* The table cursor is read only. */
+    const char *table_cursor_cfg[] = {
+      WT_CONFIG_BASE(session, WT_SESSION_open_cursor), "read_only=true", NULL};
+
+    *cursorp = NULL;
+    WT_RET(__wt_calloc_one(session, &version_cursor));
+    cursor = (WT_CURSOR *)version_cursor;
+    *cursor = iface;
+    cursor->session = (WT_SESSION *)session;
+
+    /* Open the table cursor. */
+    WT_ERR(__wt_open_cursor(session, uri, cursor, table_cursor_cfg, &version_cursor->table_cursor));
+    cursor->key_format = WT_UNCHECKED_STRING(QQQQQQBBBB);
+    cursor->value_format = version_cursor->table_cursor->value_format;
+    WT_ERR(__wt_strdup(session, uri, &cursor->uri));
+
+    /* Open the history store cursor for operations on the regular history store .*/
+    WT_ERR(__wt_curhs_open(session, cursor, &version_cursor->hs_cursor));
```

Comment:
We need to handle the case there is no history store, such as in memory database.